ksqldb-engine/src/main/java/io/confluent/ksql/serde/SerdeFeaturesFactory.java

```diff
@@ -258,6 +282,7 @@ private static void validateExplicitValueWrapping(
 
   private static boolean formatSupportsMultipleColumns(final KeyFormat format) {
     return !format.getFormat().equals(KafkaFormat.NAME)
-        && !format.getFormat().equals(NoneFormat.NAME);
+        && !format.getFormat().equals(NoneFormat.NAME)
+        && !format.getFormat().equals(DelimitedFormat.NAME);
```

Comment:
We also need to check delimited type as well

Reply:
This change is not correct. The delimited format _does_ support multiple columns, just not complex types (struct, array, map).